src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -1330,6 +1351,14 @@ void close() {
         if (debug.on()) debug.log("close stream %d", streamid);
         Log.logTrace("Closing stream {0}", streamid);
         connection.closeStream(streamid);
+        var s = responseSubscriber == null
+                ? pendingResponseSubscriber
+                : responseSubscriber;
+        if (debug.on()) debug.log("subscriber is %s", s);
+        if (s instanceof Http2StreamResponseSubscriber<?> sw) {
+            if (debug.on()) debug.log("closing response subscriber stream %s", streamid);
+            sw.streamClosed(errorRef.get());
```

Comment:
Given the number of different methods we have and the potential confusion it can cause about which one to use when, do you think in this case we could just reuse the existing ones instead of this new one? Something like:
```
if (s instanceof HttpBodySubscriberWrapper sw) {
....
  if (!sw.completed()) {
     var cause = errorRef.get();
     var ex = cause == null
            ? new IOException("stream closed")
            : cause;
     complete(ex);
  }
}

Reply:
Excellent suggestion. Done.